sinon/lib/spy.py

```diff
@@ -68,77 +69,60 @@ def __set_args_list(self, new_args_list):
         For solving special case of mock
         Args: List (new argument list)
         """
+        self.__get_wrapper().__set__("args_list", new_args_list)
+
+    def __get_wrapper(self):
+        """
+        Return:
+            Wrapper object
+        Raise:
+            Exception if wrapper object cannot be found
+        """
+        #TODO: consider to remove "PURE" condition
         if self.args_type == "MODULE_FUNCTION":
-            getattr(self.obj, self.prop).__set__("args_list", new_args_list)
+            return getattr(self.obj, self.prop)
         elif self.args_type == "FUNCTION":
-            getattr(self.g, self.obj.__name__).__set__("args_list", new_args_list)
+            return getattr(self.g, self.obj.__name__)
         elif self.args_type == "PURE":
-            getattr(self.pure, "func").__set__("args_list", new_args_list)
+            return getattr(self.pure, "func")
+        else:
+            raise Exception('Wrapper object cannot be found')
```

Comment:
https://github.com/note35/sinon/blob/master/sinon/lib/util/ErrorHandler.py

I just lost this part, I recommend to use `ErrorHandler.py` to define exception. You can define error function such as below and import it to here.

```python
def wrapper_object_is_not_found_error():
    error_msg = 'Wrapper object cannot be found'
    return __exception_helper(error_msg)
```

Then call that function here:

```python
from .util import ErrorHandler
...
else:
    ErrorHandler.wrapper_object_is_not_found_error()
...
```